src/ios/JXcoreExtension.m

```diff
@@ -28,13 +28,118 @@
 #import "JXcoreExtension.h"
 #import "THEAppContext.h"
 
+#import "JXcore.h"
+
 // JXcoreExtension implementation.
 @implementation JXcoreExtension
 
 // Defines methods.
 - (void)defineMethods
 {
-    [[THEAppContext singleton] defineJavaScriptExtensions];
+  THEAppContext *theApp = [THEAppContext singleton];
+
+  // Export the public API to node
+
+  // StartBroadcasting
+  [JXcore addNativeBlock:^(NSArray * params, NSString * callbackId) 
+  {
+    NSLog(@"jxcore: startBroadcasting");
+
+    if ([params count] != 3 || ![params[0] isKindOfClass:[NSString class]] || 
```

Comment:
Non Blocking - Typicallin in Javascript one just ignores additional arguments. So technically you only need to check if there at least 3 arguments, not exactly 3.